publisher-sdk/src/androidTest/java/com/criteo/publisher/adview/AdWebViewClientTest.kt

```diff
@@ -52,28 +59,37 @@ class AdWebViewClientTest {
 
   @Rule
   @JvmField
-  val mockedDependenciesRule = MockedDependenciesRule()
+  val mockedDependenciesRule = MockedDependenciesRule().withSpiedLogger()
 
   private val lookup = WebViewLookup()
   private val clicker = WebViewClicker()
+  private val mraidData = MraidData()
 
   @SpyBean
   private lateinit var context: Context
 
+  @MockBean
+  private lateinit var logger: Logger
+
   @Mock
   private lateinit var listener: RedirectionListener
   private lateinit var webView: WebView
   private lateinit var webViewClient: AdWebViewClient
+  private val pageFinishedLoading = CompletableFuture<Unit>()
```

Comment:
nit: Such logic are generally synchronized with [CountDownLatch](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/CountDownLatch.html) objects. But I'm not sure if they are supported by Android 🤷‍♂️

Reply:
I used [this](https://github.com/criteo/android-publisher-sdk/blob/main/publisher-sdk/src/main/java/com/criteo/publisher/util/CompletableFuture.java) implementation of CompletableFuture which uses CountDownLatch under the hood. We actually have android version of CountDownLatch and it can be used here